indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -849,6 +849,8 @@ private TaskStatus generateAndPublishSegments(
     final PartitionsSpec partitionsSpec = partitionAnalysis.getPartitionsSpec();
     final IndexTuningConfig tuningConfig = ingestionSchema.getTuningConfig();
     final long pushTimeout = tuningConfig.getPushTimeout();
+    tuningConfig.useMaxMemoryEstimates = getContextValue(
```

Comment:
If it's on the tuning config, then it can be taken as another parameter from the tuning config's JSON and doesn't need to be on the context.  Setting values on the config like this is a bit dangerous, so let's just take it on the constructor and stop looking at the context of the Task.